File: zerojudge/zerojudgea010.py
import math
primeList = []
isPrime = []
maxn = 2000000


def catchPrime():
    # init
    for i in range(maxn):
        isPrime.append(True)
    isPrime[0] = False
    isPrime[1] = False
    # faster
    sqrtMaxn = int(math.sqrt(maxn))
    # simple prime algorithm
    for i in range(2, sqrtMaxn):
        if isPrime[i]:
            for k in range(int((maxn-1)/i), i-1, -1):
                if isPrime[k]:
                    isPrime[i * k] = False
    # put into a list
    for i in range(maxn):
        if isPrime[i]:
            primeList.append(i)


catchPrime()
num = int(input())
cnt = 0
fac = primeList[cnt]
first = True
while num > 1 and fac <= num:
    d = 0
    while num > 1 and num % fac == 0:
        d += 1
        num = int(num / fac)
    if not(first):
        print(' * ', end='')
    if d > 0:
        first = False
        print(fac, end='')
        if d > 1:
            print('^' + str(d), end='')
    if num > 1 and fac <= num:
        cnt += 1
        while num % primeList[cnt] != 0:
            cnt += 1
        fac = primeList[cnt]

print()

File: zerojudge/zerojudgea010.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;
const int maxn = 2000000;
int prime[maxn];
vector<int> primevector;
void catchprime()
{
    for (int i = 0; i < maxn; i++)
        prime[i] = 1;
    prime[0] = 0;
    prime[1] = 0;
    int sqrt_maxn = sqrt(maxn);
    for (int i = 2; i <= sqrt_maxn; i++)
    {
        if (prime[i] == 1)
        {
            for (int k = (maxn - 1) / i, j = i * k; k >= i; k--, j -= i)
            {
                if (prime[k] == 1)
                {
                    prime[j] = 0;
                }
            }
        }
    }
    for (int i = 0; i < maxn; i++)
        if (prime[i] == 1)
            primevector.push_back(i);
}
int main(void)
{
    catchprime();
    int n;
    cin >> n;
    int fac = 2;
    int first = 1;
    int cnt = 0;
    while (n > 1 && fac <= n)
    {
        int d = 0;
        while (n % fac == 0 && n > 1)
        {
            d++;
            n /= fac;
        }
        if (!first)
            cout << " * ";
        if (d)
        {
            cout << fac;
            first = 0;
        }
        if (d > 1)
            cout << "^" << d;
        if (n > 1 && fac <= n)
        {
            cnt ++;
            while(n % primevector[cnt] != 0)
            {
                cnt ++;
            }
            fac = primevector[cnt];
        }
        // cout << "fac:" << fac << " n: " << n << endl;
    }
    cout << endl;
    return 0;
}